testing_test.go

```diff
@@ -47,31 +47,31 @@ func Test_isNotFound(t *testing.T) {
 		},
 		"join-other": {
 			err:  errors.Join(nil, notFoundError{}, errors.New("other")),
-			want: false,
+			want: true,
```

Comment:
@stevenh I'm not sure, but this PR unveiled this test errors 🤔 Do you think we were having a false sense of security with them? Changing the assertion fixed them, and reading the specs for the modified tests makes me think they are correct now: a joined error including notFound _should_ be captured as `want:true`, shouldn't it?